torch/utils/launch_cpu_inference.py

```diff
@@ -0,0 +1,577 @@
+                    else:
+                        os.environ["LD_PRELOAD"] = matches[0]
+                    lib_find = True
+                    break
+        return lib_set or lib_find
+
+
+    def set_memory_allocator(self, enable_tcmalloc=True, enable_jemalloc=False, use_default_allocator=False):
+        """
+        Enable TCMalloc/JeMalloc with LD_PRELOAD and set configuration for JeMalloc.
+        By default, PTMalloc will be used for PyTorch, but TCMalloc and JeMalloc can get better
+        memory resue and reduce page fault to improve performance.
+        """
+        if enable_tcmalloc and enable_jemalloc:
+            logger.error("Unable to enable TCMalloc and JEMalloc at the same time")
+            exit(-1)
+
+        if enable_tcmalloc:
+            find_tc = self.add_lib_preload(lib_type="tcmalloc")
+            if not find_tc:
+                msg = f"{msg_log_notfound} you can use \"conda install -c conda-forge gperftools\" to install {{0}}"
+                logger.warning(msg.format("TCmalloc", "tcmalloc"))
+            else:
+                logger.info("Use TCMalloc memory allocator")
+
+        elif enable_jemalloc:
+            find_je = self.add_lib_preload(lib_type="jemalloc")
+            if not find_je:
+                msg = f"{msg_log_notfound} you can use \"conda install -c conda-forge jemalloc\" to install {{0}}"
+                logger.warning(msg.format("Jemalloc", "jemalloc"))
+            else:
+                logger.info("Use JeMalloc memory allocator")
+                self.set_env("MALLOC_CONF", "oversize_threshold:1,background_thread:true,metadata_thp:auto")
+
+        elif use_default_allocator:
+            pass
+
+        else:
+            find_tc = self.add_lib_preload(lib_type="tcmalloc")
+            if find_tc:
+                logger.info("Use TCMalloc memory allocator")
+                return
+            find_je = self.add_lib_preload(lib_type="jemalloc")
+            if find_je:
+                logger.info("Use JeMalloc memory allocator")
+                return
+            logger.warning(f"""Neither TCMalloc nor JeMalloc is found in $CONDA_PREFIX/lib or $VIRTUAL_ENV/lib
+                            or /.local/lib/ or /usr/local/lib/ or /usr/local/lib64/ or /usr/lib or /usr/lib64 or
+                           {expanduser("~")}/.local/lib/ so the LD_PRELOAD environment variable will not be set. This may drop the performance""")
+
+    def logger_env(self, env_name=""):
+        if env_name in os.environ:
+            logger.info(f"{env_name}={os.environ[env_name]}")
+
+    def set_env(self, env_name, env_value=None):
+        if not env_value:
+            logger.warning(f"{env_name} is None")
+        if env_name not in os.environ:
+            os.environ[env_name] = env_value
+        elif os.environ[env_name] != env_value:
+            logger.warning(f"{env_name} in environment variable is {os.environ[env_name]} while the value you set is {env_value}")
```

Comment:
same here, if it is a valid action to set env vars that are already present this should be an "info" level log. Also the message can be improved:

```
logger.warning(f"Overriding already set environment variable: {env_name}. New value: {env_value}. Previous value: {prev_value}"
```

Reply:
We don't mean to override the environment variable. We take values that are set by environment variables as the highest priority. This just notifies the user that he/she has set values via the environment variable, which is different than that set in the script.